File: src/main/java/com/fouan/events/BoardInitialized.java
package com.fouan.events;

import com.fouan.zones.Zone;
import com.fouan.zones.view.Connection;
import lombok.Getter;

import java.util.List;
import java.util.Set;

@Getter
public final class BoardInitialized
        extends Event<BoardInitialized>
        implements ZoneEvent {

    private final List<Zone> zones;
    private final Set<Connection> connections;

    public BoardInitialized(int turn, List<Zone> zones, Set<Connection> connections) {
        super(turn);
        this.zones = List.copyOf(zones);
        this.connections = Set.copyOf(connections);
    }
}

File: src/main/java/com/fouan/events/BoardInitialized.kt
package com.fouan.events

import com.fouan.zones.Zone
import com.fouan.zones.view.Connection

class BoardInitialized(turn: Int, zones: List<Zone>, connections: Set<Connection>) :
    Event<BoardInitialized>(turn), ZoneEvent {
    val zones: List<Zone>
    val connections: Set<Connection>

    init {
        this.zones = zones.toMutableList()
        this.connections = connections.toMutableSet()
    }
}
